common/flatpak-dir.c

```diff
@@ -7982,6 +8032,80 @@ flatpak_dir_find_installed_ref (FlatpakDir   *self,
   return NULL;
 }
 
+static GPtrArray *
+filter_out_deployed_refs (FlatpakDir *self,
```

Comment:
Nitpick: It would be good to have a comment above this function summarising what it does; makes the code a bit easier to understand at a glance.